clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -349,19 +347,20 @@ private void sendProduceRequest(long now, int destination, short acks, int timeo
             recordsByPartition.put(tp, batch);
         }
 
-        ProduceRequest produceRequest = new ProduceRequest(acks, timeout, produceRecordsByPartition);
-        RequestHeader header = this.client.nextRequestHeader(ApiKeys.PRODUCE);
+        ProduceRequest.Builder requestBuilder =
+                new ProduceRequest.Builder(acks, timeout, produceRecordsByPartition);
```

Comment:
Sure, we can enforce a minimum version of 2 here.  Brokers earlier than 0.10 will automatically disconnect our clients, since those brokers don't support ApiVersionsRequest, which is the first thing we send after KAFKA-3600.